src/AstTranslator.cpp

```diff
@@ -542,10 +545,11 @@ std::unique_ptr<RamOperation> AstTranslator::ClauseTranslator::createOperation(c
         auto arity = head->getArity() - auxiliaryArity;
         std::vector<std::unique_ptr<RamExpression>> values;
         bool isVolatile = true;
+        auto args = head->getArguments();
 
         // add args for original tuple
         for (size_t i = 0; i < arity; i++) {
```

Comment:
Now that we don't need getArgument, it would be better to have something like:
```suggestion
        for (auto* arg : args) {
```

Reply:
Is it better to iterate with index? Since arity = head->getArity() - auxiliaryArity.

Reply:
Ah, I missed that `arity` was computed. You're right, don't change it